torch/csrc/cuda/comm.h

```diff
@@ -0,0 +1,14 @@
+#include "torch/csrc/assertions.h"
+
+#include <ATen/ATen.h>
+#include <unordered_map>
+
+namespace torch { namespace cuda {
+
+using tensor_list2d = std::vector<std::vector<at::Tensor>>;
+
+std::vector<at::Tensor> broadcast(const at::Tensor& tensor, const at::IntList devices);
```

Comment:
I don't think `const` has any API semantics here for `IntList` and `TensorList`. It's still `at::ArrayRef<Tensor>` not `at::ArrayRef<const Tensor>`.